File: Container With Most Water.go
func maxArea(height []int) int {
    n := len(height)
    left, right := 0, n-1
    maxx := 0
    for left < right {
        maxx = max(maxx, (right-left)*min(height[left], height[right]))
        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }
    return maxx
}

File: Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int left = 0, right = n-1;
        int maxx = 0;
        while (left < right) {
            maxx = max(maxx, (right-left)*min(height[left], height[right]));
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxx;
    }
};
